Search: Send users to the right place for customization post-checkout

a8ck3n opened the pull request:
#### Changes proposed in this Pull Request

Updates the copy to direct users to either the Search Dashboard (for Atomic) or the Search Customizer (for Simple).

#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Confirm tests are passing. Run the isolated tests with this command: `yarn run test-client client/state/sites/test/selectors.js`
* Create a Simple site and purchase a Jetpack Search plan. You'll be punted to WordPress.com for the actual checkout. Once completed, return to `http://calypso.localhost` and load the `/checkout/thank-you/DOMAIN-BITS` URL to confirm the copy has been updated and the button does the correct thing. The button should read, "Customize Search" and should load the correct customizer page.
* Create a new site, purchase a Business plan, install a plugin (to go Atomic), then add a Search plan. Again, return to local Calypso and test the `/checkout/thank-you/DOMAIN-BITS` URL to confirm the button is shown correctly. Should read "Go to Search Dashboard" and link to `/wp-admin/admin.php?page=jetpack-search`.
* Test with a self-hosted site. Should work the same as for Atomic.
* Test with JP 10.0 or earlier. In those versions, the default dashboard URL is returned (`/wp-admin/admin.php?page=jetpack`). Only on 10.1 or later do you get the Jetpack Search dashboard.

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Notes:

1. This is a test/draft PR. — Linked to issue now.
2. I had to disable the `.husky/pre-commit` and `.husky/pre-push` hooks to work on this branch. — Fixed. Issue was specific to Tower for macOS.

For Translation:

1. In the meantime, we have configured Jetpack Search on your site — try customizing it!
2. Go to Search Dashboard

![LocalizeSnap](https://user-images.githubusercontent.com/40267301/155655554-fcb91231-e745-484b-a28e-c2c67e6a5500.png)

a8ck3n committed:
Refactor test for search status.

The same test is used in three separate functions so best not to repeat ourselves.

a8ck3n committed:
Update copy for shown after purchasing a Search plan.

Just a minor change to make the message more generic. Using the same message for Atomic & Simple sites.

a8ck3n committed:
Move test for Search above My Home test.

We already has the necessary JSX in place but that code path was never hit. With this change, it should be triggered for any purchase of a Search plan.

a8ck3n commented:
When testing with either a Simple or Atomic site, the resulting page should look like this:

![SimpleResults](https://user-images.githubusercontent.com/40267301/153196579-2af24a68-4c06-45db-8cfc-9cf84d647049.png)

kangzj commented:
Awesome work @a8ck3n 

> I had to disable the .husky/pre-commit and .husky/pre-push hooks to work on this branch.

Could you please share why this is happening? Should we address the problem rather than disabling the checks?

We would usually link the issue in the PR descrption, in this case, we'd add something like: `Fixes #60744`  - when the PR is closed, the issue would then be closed together.

As per the issue, for atomic sites the URL should link to the dashboard,  something like `/wp-admin/admin.php?page=jetpack-search`. I don't see that in the selector:

https://github.com/Automattic/wp-calypso/blob/bd1064d9f3b41e997cb17eca819b5985c5e6a5dd/client/state/sites/selectors/get-jetpack-search-customize-url.js#L19

Keep up the good work! 👍

bluefuton commented on client/my-sites/checkout/checkout-thank-you/header.jsx:
```diff
@@ -55,8 +55,13 @@ export class CheckoutThankYouHeader extends PureComponent {
 		recordStartTransferClickInThankYou: PropTypes.func.isRequired,
 	};
 
+	isSearch() {
```
Nice cleanup 👍

a8ck3n committed:
Factor out logic for the primary button.

We update the button title and the target URL based on if theh site is a Jetpack site or not. Currently returns a placeholder URL for the “Go to Search Dashboard” button.

a8ck3n committed:
Add some initial tests around new selector.

Two tests for getJetpackSearchDashboardUrl.

1. Confirm null when site not loaded.
2. Confirm ‘#’ when we have a Jetpack site.

a8ck3n committed:
Implement getJetpackSearchDashboardUrl.

With tests to confirm expected behaviour.

1. Return null when site not loaded.
2. Return null for Simple sites.
3. Return null if adminUrl is not found.
4. Return default dashboard for versions prior to 10.1.
5. Return Search dashboard for vesion 10.1 and later.

a8ck3n commented:
> Could you please share why this is happening? Should we address the problem rather than disabling the checks?

So turns out the commit hooks were working on the command line but failing for me in Tower. Was complaining that `yarn` command was unknown. Creating a `.huskyrc` file in my home directory and exporting my PATH fixed the issue.

a8ck3n commented:
Two quick videos showing results of changes. For Simple Sites:

https://user-images.githubusercontent.com/40267301/154477485-a619b531-359a-4c5b-98c7-aa758a64e17d.mp4

And for Atomic Sites:

https://user-images.githubusercontent.com/40267301/154477533-da6e280f-1d1e-48cd-827d-8335245f9e89.mp4

bluefuton commented on client/state/sites/test/selectors.js:
```diff
@@ -2600,6 +2601,96 @@ describe( 'selectors', () => {
+				2916284
+			);
+
+			chaiExpect( dashboardUrl ).to.be.null;
```
We're using Jest assertions these days rather than Chai in new tests. You can easily change them:

`expect( dashboardUrl ).toBeNull();`

a8ck3n replied:
Tests have been updated!

bluefuton commented on client/my-sites/checkout/checkout-thank-you/header.jsx:
```diff
@@ -96,22 +104,23 @@ export class CheckoutThankYouHeader extends PureComponent {
 
-		if ( purchases?.length > 0 && purchases[ 0 ].productType === 'search' ) {
+		if ( this.isSearch() ) {
+			// Handle the Atomic vs Simple division here if needed.
```
I'd be fine to use the same copy for both cases 👍

a8ck3n committed:
Updated tests to use Jest.

a8ck3n committed:
Remove unneeded comment.

kangzj commented on client/state/sites/selectors/get-jetpack-search-dashboard-url.js:
```diff
@@ -0,0 +1,27 @@
+	if ( jetpackVersion && versionCompare( jetpackVersion, '10.1', '>=' ) ) {
+		return adminUrl + 'admin.php?page=jetpack-search';
+	}
+	return adminUrl + 'admin.php?page=jetpack';
```
Could we link to performance page instead, where users could see toggles and link to customizer - even the page doesn't exist, it would be on dashboard as well.
`/wp-admin/admin.php?page=jetpack#/performance`

a8ck3n replied:
Can do. 👍

kangzj reviewed (commented):
Simple sites and Atomic work as described. Great job @a8ck3n 👍 

However Jetpack sites are redirected straight away to the site with URL `/wp-admin/admin.php?page=jetpack`, I think which is fine. It'll be better to change the redirection URL to `/wp-admin/admin.php?page=jetpack#/performance`, to align with Atomic sites. This might not fall in the scope of the PR tho.

bluefuton commented:
Let's give this a rebase - I think that should solve the E2E test issues 🤞

a8ck3n committed:
Update URL for older versions of Jetpack.

If not on 10.1 or later, send the user to the following:

/admin.php?page=jetpack#/performance

a8ck3n committed:
Add getJetpackSearchDashboardUrl selector.

Just wired things up. Will be used to return the URL of the Jetpack Search Dashboard. Currently returns ‘#’.

bluefuton commented:
I set out to test with JP 10.0 and a Jurassic Ninja site, and interestingly I get the generic thank you message rather than a Jetpack Search one:

<img width="1028" alt="Screen Shot 2022-02-24 at 15 40 21" src="https://user-images.githubusercontent.com/17325/155448033-dc377b85-5d4d-4f2f-80ba-f1ff2352719b.png">

I set a breakpoint to see if `isSearch` was being called, and `purchases` seems to be an empty array:

<img width="613" alt="Screen Shot 2022-02-24 at 15 47 02" src="https://user-images.githubusercontent.com/17325/155448309-0e631a96-6431-463b-902e-777cc45b9261.png">

I wonder why that's happening 🤔 I bought with credits but I thought it'd still count as a purchase. The site in question is:

http://calypso.localhost:3000/checkout/thank-you/tipped-turtle.jurassic.ninja

Store admin shows that I have a current subscription to Jetpack Search.

bluefuton commented:
Ah - @kangzj helped me out. The URL needs the receipt ID at the end, like:

http://calypso.localhost:3000/checkout/thank-you/tipped-turtle.jurassic.ninja/1234567

bluefuton reviewed (approved):
Tested with Jetpack 10.0 and can confirm you're directed to `/wp-admin/admin.php?page=jetpack#/performance`.

🚢

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7155355

Hi @a8ck3n, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `In the meantime, we have configured Jetpack Search on your site — try customizing it!`
- `Go to Search Dashboard`

Thank you in advance!

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7155355).
